libc/stdio_scanf.c

```diff
@@ -0,0 +1,3510 @@
+
+	TEST_ASSERT_EQUAL_INT(1, vsscanfWrapper(buff, "%[^s]", chr));
+	TEST_ASSERT_EQUAL_STRING("Loremip", chr);
+
+	TEST_ASSERT_EQUAL_INT(1, sscanf(buff, "%[^s]", chr));
+	TEST_ASSERT_EQUAL_STRING("Loremip", chr);
+
+	rewind(rd);
+
+	/* 
+	 *.Scanf will read "rd" until elements between brackets and "rd" match
+	 * If it finds incompatibility between them it will stop at the point where it occurred
+	 */
+	TEST_ASSERT_EQUAL_INT(1, vfscanfWrapper(rd, "%[Lori]", chr));
+	TEST_ASSERT_EQUAL_STRING("Lor", chr);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(1, fscanf(rd, "%[Lori]", chr));
+	TEST_ASSERT_EQUAL_STRING("Lor", chr);
+
+	TEST_ASSERT_EQUAL_INT(1, vsscanfWrapper(buff, "%[Lori]", chr));
+	TEST_ASSERT_EQUAL_STRING("Lor", chr);
+
+	TEST_ASSERT_EQUAL_INT(1, sscanf(buff, "%[Lori]", chr));
+	TEST_ASSERT_EQUAL_STRING("Lor", chr);
+}
+
+
+TEST(stdio_scanf_squareBrackets, squareBrackets_extended)
+{
+	char buff[BUFF_LEN_STR] = "Loremipsumdolorsit";
+	char chr[BUFF_LEN_STR] = { 0 };
+	char chr2[BUFF_LEN_STR] = { 0 };
+
+	fprintf(rd, "%s", buff);
+	rewind(rd);
+
+	/* 
+	 * Scanf will read the string and match word "psu" as a point of stop and save this is "chr"
+	 * After that it will set the file position indicator before "psu"
+	 * anything else after that will be saved to "chr2"
+	 */
+	TEST_ASSERT_EQUAL_INT(2, vfscanfWrapper(rd, "%[^psu]%s", chr, chr2));
+	TEST_ASSERT_EQUAL_STRING("Loremi", chr);
+	TEST_ASSERT_EQUAL_STRING("psumdolorsit", chr2);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(2, fscanf(rd, "%[^psu]%s", chr, chr2));
+	TEST_ASSERT_EQUAL_STRING("Loremi", chr);
+	TEST_ASSERT_EQUAL_STRING("psumdolorsit", chr2);
+
+	TEST_ASSERT_EQUAL_INT(2, vsscanfWrapper(buff, "%[^psu]%s", chr, chr2));
+	TEST_ASSERT_EQUAL_STRING("Loremi", chr);
+	TEST_ASSERT_EQUAL_STRING("psumdolorsit", chr2);
+
+	TEST_ASSERT_EQUAL_INT(2, sscanf(buff, "%[^psu]%s", chr, chr2));
+	TEST_ASSERT_EQUAL_STRING("Loremi", chr);
+	TEST_ASSERT_EQUAL_STRING("psumdolorsit", chr2);
+
+	rewind(rd);
+	memset(chr, 0, sizeof(chr));
+	memset(chr2, 0, sizeof(chr2));
+
+	/* 
+	 * Scanf will try to find elements after "Lor" and save them to "chr" until first
+	 * occurrence of "do" everything after that with content of [] will be saved to "chr2"
+	*/
+	TEST_ASSERT_EQUAL_INT(2, vfscanfWrapper(rd, "Lor%[^do]%s", chr, chr2));
+	TEST_ASSERT_EQUAL_STRING("emipsum", chr);
+	TEST_ASSERT_EQUAL_STRING("dolorsit", chr2);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(2, fscanf(rd, "Lor%[^do]%s", chr, chr2));
+	TEST_ASSERT_EQUAL_STRING("emipsum", chr);
+	TEST_ASSERT_EQUAL_STRING("dolorsit", chr2);
+
+	TEST_ASSERT_EQUAL_INT(2, vsscanfWrapper(buff, "Lor%[^do]%s", chr, chr2));
+	TEST_ASSERT_EQUAL_STRING("emipsum", chr);
+	TEST_ASSERT_EQUAL_STRING("dolorsit", chr2);
+
+	TEST_ASSERT_EQUAL_INT(2, sscanf(buff, "Lor%[^do]%s", chr, chr2));
+	TEST_ASSERT_EQUAL_STRING("emipsum", chr);
+	TEST_ASSERT_EQUAL_STRING("dolorsit", chr2);
+}
+
+
+TEST(stdio_scanf_squareBrackets, squareBrackets_spaces)
```

Comment:
Why spaces? I don't see any spaces there. To be honest I don't see anything new comparing to the previous test case.